lightning/src/ln/channelmanager.rs

```diff
@@ -2839,17 +2844,40 @@ where
 			}
 		};
 		if let Some(update) = update_opt {
-			// Try to send the `BroadcastChannelUpdate` to the peer we just force-closed on, but if
-			// not try to broadcast it via whatever peer we have.
-			let per_peer_state = self.per_peer_state.read().unwrap();
-			let a_peer_state_opt = per_peer_state.get(peer_node_id)
-				.ok_or(per_peer_state.values().next());
-			if let Ok(a_peer_state_mutex) = a_peer_state_opt {
-				let mut a_peer_state = a_peer_state_mutex.lock().unwrap();
-				a_peer_state.pending_msg_events.push(events::MessageSendEvent::BroadcastChannelUpdate {
-					msg: update
-				});
-			}
+			PersistenceNotifierGuard::optionally_notify(self, || {
+				let mut should_persist = NotifyOption::SkipPersistNoEvents;
+
+				// Try to send the `BroadcastChannelUpdate` to the peer we just force-closed on, but if
+				// not try to broadcast it via whatever peer we have.
+				let brodcast_message_evt = events::MessageSendEvent::BroadcastChannelUpdate {
+					msg: update.clone()
+				};
+
+				let per_peer_state = self.per_peer_state.read().unwrap();
+
+				// Attempt to get the a_peer_state_mutex for the peer we disconnected on.
+				let a_peer_state_mutex_opt = per_peer_state.get(peer_node_id).map(|v| v);
+
+				// If the particular peer is not present. Select any random peer from the ones we are connected on.
+				let a_peer_state_mutex_opt = a_peer_state_mutex_opt.or_else(|| per_peer_state.iter().next().map(|(_, v)| v));
```

Comment:
Make sense
Since we will only be able to broadcast to peers we are connected with, it makes to not add the broadcast message to pending_msg_evts queues of disconnected peers.

Addressed in [40bbfeb](https://github.com/lightningdevkit/rust-lightning/pull/2731/commits/40bbfeb6f36e0b25d93d816c0cc1db84b791960e)
Thanks for the suggestion!